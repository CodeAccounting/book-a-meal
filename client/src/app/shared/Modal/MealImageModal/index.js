import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import MealImageModal from './MealImageModal';
import { editMeal } from '../../../../store/operations/meals';
import { getLastMealItem } from '../../../../store/selectors/meals';

const mapStateToProps = state => ({
  updating: state.meals.working,
  mealId: getLastMealItem(state.meals.items).id,
  formerImgURL: getLastMealItem(state.meals.items).imageURL
});

const mapDispatchToProps = dispatch => bindActionCreators({
  editMeal
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(MealImageModal);
