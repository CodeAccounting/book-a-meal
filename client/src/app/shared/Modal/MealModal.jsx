import React from 'react';
import PropTypes from 'prop-types';
import Form from '../Form';
import { mealObjPropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * @function MealModal
 * @param {object} props
 * @returns {component} AddMealModal
 */
const MealModal = (props) => {
  const { type } = props;
  return <Form {...props} type={type} meta={{ btnText: 'ADD MEAL' }} />;
};

MealModal.propTypes = {
  type: PropTypes.string.isRequired,
  meal: mealObjPropTypes
};

MealModal.defaultProps = {
  meal: null
};

export default MealModal;
