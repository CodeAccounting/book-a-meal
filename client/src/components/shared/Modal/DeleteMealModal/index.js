import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import DeleteMealModal from './DeleteMealModal';
import { deleteMeal } from '../../../../store/operations/meals';

const mapStateToProps = state => ({
  deleting: state.meals.working
});

const mapDispatchToProps = dispatch => bindActionCreators({
  deleteMeal
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(DeleteMealModal);
