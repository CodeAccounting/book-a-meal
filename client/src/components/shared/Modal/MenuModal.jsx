import React from 'react';
import MenuForm from '../Form/MenuForm';

/**
 * @exports
 * @function MenuModal
 * @returns {component} MenuModal
 */
const MenuModal = () => <MenuForm />;

export default MenuModal;
