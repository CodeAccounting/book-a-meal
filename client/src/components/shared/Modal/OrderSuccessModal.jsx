import React from 'react';
/**
 * @exports
 * @function OrderSuccessModal
 * @returns {component} OrderSuccessModal
 */
const OrderSuccessModal = () => <p className="text-center">Thank you for your order. Your belly will be filled up shortly</p>;

export default OrderSuccessModal;
