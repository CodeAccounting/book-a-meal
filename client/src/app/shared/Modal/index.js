import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Modal from './Modal';
import { toggleModal } from '../../../store/actions/ui';

const mapStateToProps = state => ({
  isOpen: state.ui.modals.isOpen,
  type: state.ui.modals.type
});

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleModal
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Modal);
