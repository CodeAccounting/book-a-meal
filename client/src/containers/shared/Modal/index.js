import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Modal from '../../../components/shared/Modal';
import { toggleModal } from '../../../actions/ui';

const mapStateToProps = state => ({
  open: state.ui.modals.open,
  type: state.ui.modals.type
});

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleModal
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Modal);
