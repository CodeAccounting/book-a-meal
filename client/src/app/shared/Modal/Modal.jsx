import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MealModal from './MealModal';
import MealImageModal from './MealImageModal';
import LinkBtn from '../Link';

/**
 * @exports
 * @class Modal
 * @extends Component
 * @returns {JSX} Modal Component
 */
class Modal extends Component {
  static propTypes = {
    type: PropTypes.string,
    isOpen: PropTypes.bool.isRequired,
    toggleModal: PropTypes.func.isRequired
  };

  static defaultProps = {
    type: null
  };

  /**
   * @memberof Modal
   * @param {object} props
   * @returns {null} null
   */
  static getDerivedStateFromProps(props) {
    return { bodyStyle: props.isOpen === false ? 'visible' : 'hidden' };
  }

  state = {
    bodyStyle: 'visible'
  }

  /**
     * @memberof Modal
     *@returns {JSX} Modal Component
   */
  render() {
    const { type, isOpen, toggleModal } = this.props;

    document.body.style.overflow = this.state.bodyStyle;

    if (isOpen) {
      return (
        <div className="modal show" id="meal-modal">
          <div className="modal-content">
            <div className="modal-header">
              <div className="modal-title">
                <h3 id="modal-title-h3">
                  {type === 'addMeal' && 'Add a Meal'}
                  {type === 'editMeal' && 'Edit Meal'}
                  {type === 'newMealImage' && 'Add a Meal Image'}
                </h3>
              </div>
              <div className="close">
                <LinkBtn aria-hidden="true" id="modal-close-icon" clickHandler={toggleModal}>&times;</LinkBtn>
              </div>
            </div>
            <hr />
            <div className="modal-body">
              {type === 'addMeal' && <MealModal {...this.props} type="addMeal" />}
              {type === 'newMealImage' && <MealImageModal {...this.props} type="newMealImage" />}
            </div>
          </div>
        </div>
      );
    }

    return null;
  }
}

export default Modal;
