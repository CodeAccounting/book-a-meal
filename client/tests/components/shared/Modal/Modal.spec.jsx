import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { initialValues } from '../../../setup/data';
import Modal from '../../../../src/app/shared/Modal/Modal';
import ConnectedModal from '../../../../src/app/shared/Modal';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore(initialValues);

describe('Modal', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const shallowWrapper = shallow(<Modal type="addMeal" open toggleModal={jest.fn()} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('#modal-title-h3').text()).toEqual('Add a Meal');
    expect(shallowWrapper.find('MealModal')).toBeTruthy();
  });

  it('does not correctly when modal is closed', () => {
    const shallowWrapper = shallow(<Modal type="addMeal" open={false} toggleModal={jest.fn()} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.text()).toEqual('');
  });

  it('renders correctly when type is newMealImage', () => {
    const shallowWrapper = shallow(<Modal type="newMealImage" open toggleModal={jest.fn()} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('#modal-title-h3').text()).toEqual('Add a Meal Image');
    expect(shallowWrapper.find('MealImageModal')).toBeTruthy();

    shallowWrapper.find('#modal-close-icon').simulate('click');
  });

  it('renders null when type is unknown', () => {
    const shallowWrapper = shallow(<Modal type="unknown" open toggleModal={jest.fn()} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('calls toggleModal function prop when close modal icon is clicked', () => {
    const toggleMock = jest.fn();
    const shallowWrapper = shallow(<Modal type="addMeal" open toggleModal={toggleMock} />);

    shallowWrapper.find('Link#modal-close-icon').dive('button').simulate('click');

    expect(toggleMock).toHaveBeenCalled();
  });

  it('renders connected component correctly', () => {
    const comp = (<Provider store={store}><ConnectedModal type="addMeal" open toggleModal={jest.fn()} /></Provider>);
    const mountedWrapper = mount(comp);

    expect(toJson(mountedWrapper)).toMatchSnapshot();
    mountedWrapper.unmount();
  });
});
