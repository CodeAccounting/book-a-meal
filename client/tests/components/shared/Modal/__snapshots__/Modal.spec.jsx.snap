// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal does not correctly when modal is closed 1`] = `""`;

exports[`Modal renders connected component correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Modal)
    isOpen={true}
    toggleModal={[MockFunction]}
    type="addMeal"
  >
    <Modal
      isOpen={false}
      toggleModal={[Function]}
      type={null}
    />
  </Connect(Modal)>
</Provider>
`;

exports[`Modal renders correctly 1`] = `
<div
  className="modal show"
  id="meal-modal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div
        className="modal-title"
      >
        <h3
          id="modal-title-h3"
        >
          Add a Meal
        </h3>
      </div>
      <div
        className="close"
      >
        <Link
          aria-hidden="true"
          className={null}
          clickHandler={[MockFunction]}
          id="modal-close-icon"
        >
          ×
        </Link>
      </div>
    </div>
    <hr />
    <div
      className="modal-body"
    >
      <MealModal
        isOpen={true}
        meal={null}
        toggleModal={[MockFunction]}
        type="addMeal"
      />
    </div>
  </div>
</div>
`;

exports[`Modal renders correctly when type is editMeal 1`] = `
<div
  className="modal show"
  id="meal-modal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div
        className="modal-title"
      >
        <h3
          id="modal-title-h3"
        >
          Edit Meal
        </h3>
      </div>
      <div
        className="close"
      >
        <Link
          aria-hidden="true"
          className={null}
          clickHandler={[MockFunction]}
          id="modal-close-icon"
        >
          ×
        </Link>
      </div>
    </div>
    <hr />
    <div
      className="modal-body"
    />
  </div>
</div>
`;

exports[`Modal renders correctly when type is newMealImage 1`] = `
<div
  className="modal show"
  id="meal-modal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div
        className="modal-title"
      >
        <h3
          id="modal-title-h3"
        >
          Add a Meal Image
        </h3>
      </div>
      <div
        className="close"
      >
        <Link
          aria-hidden="true"
          className={null}
          clickHandler={[MockFunction]}
          id="modal-close-icon"
        >
          ×
        </Link>
      </div>
    </div>
    <hr />
    <div
      className="modal-body"
    >
      <Connect(MealImageModal)
        isOpen={true}
        toggleModal={[MockFunction]}
        type="newMealImage"
      />
    </div>
  </div>
</div>
`;
